flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBConfigurableOptions.java

```diff
@@ -90,7 +90,7 @@
     public static final ConfigOption<String> LOG_DIR =
             key("state.backend.rocksdb.log.dir")
                     .stringType()
-                    .noDefaultValue()
+                    .defaultValue(getFlinkLogDir())
```

Comment:
I don't think this is a correct implementation. The configuration of `state.backend.rocksdb.log.dir` is shown in [rocksdb configuration doc](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/config/#state-backend-rocksdb-log-dir). This should not be a null here. Moreover, the value should be analyzed when launching RocksDB state-backend instead of static method initialization.

I think we can mark it as `noDefaultValue` but give correct description.